### PR TITLE
[NT-1704] Update ReactiveSwift to 6.5.0

### PR DESCRIPTION
**Pull request by justinswart**
# 📲 What

Updates ReactiveSwift to `6.5.0`.

**Note:** https://github.com/kickstarter/Kickstarter-ReactiveExtensions/pull/82 should merge before this.

# 🤔 Why

There have been a number of improvements to this framework that we use throughout our codebase (see release notes below).

# 🛠 How

- Updated our `Cartfile`.
- Replaced deprecated instances of `filterMap` with `compactMap`.
- Updated CircleCI to use Xcode 12.2 and iOS 14.2.
- Fixed Carthage workaround script.

# 👀 See

[ReactiveSwift release notes](https://github.com/ReactiveCocoa/ReactiveSwift/releases)
[Carthage workaround](https://github.com/Carthage/Carthage/blob/master/Documentation/Xcode12Workaround.md)

# ✅ Acceptance criteria

- [ ] Build and tests pass

### CODE REVIEW

File: Cartfile
Comment (justinswart) [inline Cartfile:4]: This SHA will be updated once https://github.com/kickstarter/Kickstarter-ReactiveExtensions/pull/82 merges.

File: bin/carthage.sh
Comment (justinswart) [inline bin/carthage.sh:15]: These were no longer needed for the workaround and caused Alamofire to fail to build.

**Comment by justinswart** [comment]
@singhhari blocking this until #1352 merges - it appears that iOS 14.2 may require us to re-record all of our snapshots 🤦‍♂️

File: Library/ViewModels/DiscoveryPageViewModel.swift
Comment (singhhari) [inline Library/ViewModels/DiscoveryPageViewModel.swift:401]: Interesting. The update to 6.5 wants the value param explicitly written now?

**Review by singhhari — APPROVED** [review]
lgtm

File: Library/ViewModels/DiscoveryPageViewModel.swift
Comment (justinswart) [reply Library/ViewModels/DiscoveryPageViewModel.swift:401]: It works without it but complains about ambiguity.

**Pull request merged**
